implementation/src/test/java/io/smallrye/mutiny/operators/MultiGroupTest.java

```diff
@@ -1037,4 +1040,45 @@ public void rejectGroupByBadPrefetch() {
         assertThrows(IllegalArgumentException.class,
                 () -> Multi.createFrom().range(1, 10).group().by(i -> i % 2, 0L));
     }
+
+    @Test
+    void testUpstreamRequestsNotBlownOutOfProportion() {
+        ExecutorService executor = Executors.newFixedThreadPool(10);
```

Comment:
Should we wrap in a `try/finally` and shut the executor down?

I'm sure the whole tests suite has this minor problem, so we may address it separately.